dali/operators/image/distortion/jpeg_compression_distortion_op_cpu.cc

```diff
@@ -56,6 +58,36 @@ class JpegCompressionDistortionCPU : public JpegCompressionDistortion<CPUBackend
   std::vector<ThreadCtx> thread_ctx_;
 };
 
+template <typename Span>
+static auto layoutShape(TensorLayout layout, Span shape) {
+  std::array<Index, 4> result = {1, 1, 1, 1};
+  int d = 0;
+  if (layout.find('F') == 0) {
+    result[0] = shape[d++];
+  }
+  DALI_ENFORCE(shape.size() - d == 3);
+  DALI_ENFORCE(layout[d] == 'H');
+  result[1] = shape[d++];
+  DALI_ENFORCE(layout[d] == 'W');
+  result[2] = shape[d++];
+  DALI_ENFORCE(layout[d] == 'C');
```

Comment:
Food for thoughts - maybe you can add a relevant error message to each enforce?

Reply:
Hopefully done, dunno